node/full.go

```diff
@@ -147,6 +150,16 @@ func (qm *QitmeerFull) RegisterAccountService(cfg *config.Config) error {
 	return nil
 }
 
+func (qm *QitmeerFull) RegisterWalletService(cfg *config.Config, evm *meer.MeerChain, _am *acct.AccountManager, _tm *tx.TxManager, autoCollectOp chan types.AutoCollectUtxo) error {
+	// account manager
+	walletmgr, err := wallet.New(cfg, evm, _am, _tm, autoCollectOp)
```

Comment:
same reason, you need not to passing the channel as parameter, the only thing you have to concern is when and where you create the channel and how to share it between components.